### PR TITLE
Fix ONNX operator_export_type on the new registry

### LINKED ISSUE #87313: Improper model conversion from PyTorch to ONNX with torch.onnx.OperatorExportTypes.ONNX_ATEN flag
### 🐛 Describe the bug

I have spotted improper model conversion to ONNX when using `torch.onnx.OperatorExportTypes.ONNX_ATEN` flag with `PyTorch >=1.12.0`

My understanding from the [documentation](https://pytorch.org/docs/stable/onnx.html#aten-operators)

> OperatorExportTypes.ONNX_ATEN: All ATen ops (in the TorchScript namespace “aten”) are exported as ATen ops (in opset domain “org.pytorch.aten”). [ATen](https://pytorch.org/cppdocs/#aten) is PyTorch’s built-in tensor library, so this instructs the runtime to use PyTorch’s implementation of these ops.

is that when one uses `torch.onnx.OperatorExportTypes.ONNX_ATEN` flag, all ATen operators has to be exported to ONNX graph. When I run the code snippet below: 

```python
import torch


class ModelWithAtenFmod(torch.nn.Module):
    def forward(self, x, y):
        return torch.fmod(x, y)


x = torch.randn(3, 4, dtype=torch.float32)
y = torch.randn(3, 4, dtype=torch.float32)
OUT = torch.onnx.export_to_pretty_string(
    ModelWithAtenFmod(), (x, y),
    add_node_names=False,
    do_constant_folding=False,
    operator_export_type=torch.onnx.OperatorExportTypes.ONNX_ATEN)
print(OUT)
```
with `PyTorch 1.11.0` I obtain ONNX graph with ATen operator that is correct:
```
### PyTorch 1.11.0

ModelProto {
  producer_name: "pytorch"
  domain: ""
  doc_string: ""
  graph:
    GraphProto {
      name: "torch-jit-export"
      inputs: [{name: "aten::ATen_0", type:Tensor dtype: 1, Tensor dims: 3 4},{name: "aten::ATen_1", type:Tensor dtype: 1, Tensor dims: 3 4}]
      outputs: [{name: "2", type:Tensor dtype: 1, Tensor dims: ? ?}]
      value_infos: []
      initializers: []
      nodes: [
        Node {type: "ATen", inputs: [aten::ATen_0,aten::ATen_1], outputs: [2], attributes: [{ name: 'operator', type: string, value: 'fmod'}]}
      ]
    }
  opset_import: [OperatorSetIdProto { domain: , version: 9}OperatorSetIdProto { domain: org.pytorch.aten, version: 1}],
}
```

However, when I run the same code with `PyTorch 1.12.1`, ATen operator got substituted with ONNX native operator and that is not expected behaviour:
```
### PyTorch 1.12.1

ModelProto {
  producer_name: "pytorch"
  domain: ""
  doc_string: ""
  graph:
    GraphProto {
      name: "torch_jit"
      inputs: [{name: "onnx::Mod_0", type:Tensor dtype: 1, Tensor dims: 3 4},{name: "onnx::Mod_1", type:Tensor dtype: 1, Tensor dims: 3 4}]
      outputs: [{name: "2", type:Tensor dtype: 1, Tensor dims: 3 4}]
      value_infos: []
      initializers: []
      nodes: [
        Node {type: "Mod", inputs: [onnx::Mod_0,onnx::Mod_1], outputs: [2], attributes: [{ name: 'fmod', type: int, value: 1}]}
      ]
    }
  opset_import: [OperatorSetIdProto { domain: , version: 13}],
}
```

### Versions

PyTorch version: 1.12.1
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 12.6 (arm64)
GCC version: Could not collect
Clang version: 14.0.6
CMake version: version 3.24.1
Libc version: N/A

Python version: 3.8.13 | packaged by conda-forge | (default, Mar 25 2022, 06:04:14)  [Clang 12.0.1 ] (64-bit runtime)
Python platform: macOS-12.6-arm64-arm-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.23.4
[pip3] torch==1.12.1
[pip3] torchvision==0.13.1
[conda] numpy                     1.23.4           py38h09ac2d9_0    conda-forge
[conda] pytorch                   1.12.1                  py3.8_0    pytorch
[conda] torchvision               0.13.1                 py38_cpu    pytorch

### PR DESCRIPTION
**Pull request by thiagocrepaldi**
Fixes #87313 

Pytorch's ONNX pipelines do not run with `BUILD_CAFFE2=0`, so tests for `operator_export_type==ONNX_ATEN` or `ONNX_ATEN_FALLBACK` are not fully tested, which caused this regression to the core feature of ONNX exporter: ONNX ATen fallback.

This PR fixes this regression, but we discuss with Meta on how to use only `BUILD_CAFFE2=0` builds instead of ``BUILD_CAFFE2=1`

### CODE REVIEW

File: torch/onnx/utils.py [hunk 1]
```diff
@@ -1743,13 +1743,14 @@ def _should_aten_fallback(
...
 ):
+    namespace, _ = name.split("::")
     is_exportable_aten_op = registration.registry.is_registered_op(name, opset_version)
     is_onnx_aten_export = operator_export_type == _C_onnx.OperatorExportTypes.ONNX_ATEN
     is_aten_fallback_export = (
         operator_export_type == _C_onnx.OperatorExportTypes.ONNX_ATEN_FALLBACK
     )
-    return is_onnx_aten_export or (
-        not is_exportable_aten_op and is_aten_fallback_export
+    return namespace == "aten" and (
```
Comment (justinchuby) [inline torch/onnx/utils.py:1752]: I would simply do name.startswith("aten::")

File: torch/onnx/utils.py [hunk 2]
```diff
@@ -1861,6 +1880,7 @@ def _run_symbolic_function(
         if symbolic_function_group is not None:
             symbolic_fn = symbolic_function_group.get(opset_version)
             if symbolic_fn is not None:
+                # TODO(justinchuby): Wrap almost identical attrs assignment or comment the difference.
```
Comment (justinchuby) [inline torch/onnx/utils.py:1883]: Hi! I don't have context to this TODO and thus am not fitting to be the contact person. I suggest creating an issue for tracking, or alternatively using your github alias.

File: torch/onnx/utils.py [hunk 3]
```diff
@@ -1844,6 +1845,24 @@ def _run_symbolic_function(
         env=env,
     )
 
+    # when domain is "aten" and operator_export_type = ONNX_ATEN or ONNX_ATEN_FALLBACK
+    # create org.pytorch.aten::ATen oeprator regardless of symbolics exist or not
+    if _should_aten_fallback(ns_op_name, opset_version, operator_export_type):
```
Comment (justinchuby) [inline torch/onnx/utils.py:1850]: I can see we want the behavior for `ONNX_ATEN`. Do we want this behavior for `ONNX_ATEN_FALLBACK` as well? Should it try to translate first before falling back to Aten?

File: torch/onnx/utils.py (hunk 2 above)
Comment (justinchuby) [reply torch/onnx/utils.py:1883]: That said I am happy to do the actual work of it. It's just that whatever alias written here needs to be where the information is, as a contact.

File: torch/onnx/utils.py (hunk 3 above)
Comment (thiagocrepaldi) [reply torch/onnx/utils.py:1850]: outdated code/comment. `ONNX_ATEN_FALLBACK` is not handled here nor changed by this PR

File: torch/onnx/utils.py (hunk 2 above)
Comment (thiagocrepaldi) [reply torch/onnx/utils.py]: No worries, as the author of the [new registry](https://github.com/pytorch/pytorch/pull/84382) who heavily reimplemented/refactored this code, I was expecting you to be the most fitting and own it. Nonetheless, I've unassigned the comment.

**Review by titaiwangms — APPROVED** [review]

File: torch/onnx/utils.py [hunk 4]
```diff
@@ -1844,6 +1844,30 @@ def _run_symbolic_function(
         env=env,
     )
 
+    # Direct ATen export requested
+    # For BUILD_CAFFE2=0 builds, if domain=="aten" and operator_export_type==ONNX_ATEN,
+    #   an aten::ATen operator is created regardless of symbolics existence
+    # For BUILD_CAFFE2=1, the same applies only if there is no symbolic available
```
Comment (BowenBao) [inline torch/onnx/utils.py:1850]: Related to the CI failure on `test_py_diagnose_emits_error`, where it expects a `operator_supported_in_newer_opset_version` error for `aten::diagonal` when exporting in opset 9. After this change, the error was not thrown, and `aten::diagonal` is exported as `aten::ATen(...)`. The `operator_export_type` of `export` is the default `_C_onnx.OperatorExportTypes.ONNX`.

Is this behavior expected, for BUILD_CAFFE2=1?

**Comment by BowenBao** [comment]
Some test failures
```
2022-10-26T02:05:53.1811889Z =========================== short test summary info ============================
2022-10-26T02:05:53.1812306Z FAILED test/onnx/test_pytorch_onnx_caffe2_quantized.py::TestQuantizedOps::test_cat
2022-10-26T02:05:53.1815203Z FAILED test/onnx/test_pytorch_onnx_no_runtime.py::TestONNXExport::test_export_dict
2022-10-26T02:05:53.1816998Z FAILED test/onnx/test_pytorch_onnx_caffe2_quantized.py::TestQuantizedOps::test_qconv_model
2022-10-26T02:05:53.1818106Z FAILED test/onnx/test_pytorch_onnx_caffe2_quantized.py::TestQuantizedOps::test_qlinear_model
2022-10-26T02:05:53.1818678Z FAILED test/onnx/test_pytorch_onnx_caffe2_quantized.py::TestQuantizedOps::test_quantized_add
2022-10-26T02:05:53.1821321Z FAILED test/onnx/internal/test_diagnostics.py::TestOnnxDiagnostics::test_py_diagnose_emits_error
2022-10-26T02:05:53.1822149Z FAILED test/onnx/test_pytorch_onnx_caffe2_quantized.py::TestQuantizedOps::test_sequential
2022-10-26T02:05:53.1832674Z FAILED test/onnx/test_pytorch_onnx_caffe2_quantized.py::TestQuantizedOps::test_small_model
```

File: torch/onnx/utils.py (hunk 4 above)
Comment (thiagocrepaldi) [reply torch/onnx/utils.py:1850]: running 

```python
def test_py_diagnose_emits_error():
    class M(torch.nn.Module):
        def forward(self, x):
            return torch.diagonal(x)

    torch.onnx.export(
        M(),
        torch.randn(3, 4),
        io.BytesIO(),
        opset_version=9,
        verbose=True
    )

test_py_diagnose_emits_error()
```

on a build with `BUILD_CAFFE2=1` locally I got

```bash
Traceback (most recent call last):
  File "/pytorch_repros/bad_aten_mode.py", line 57, in <module>
    test_py_diagnose_emits_error()
  File "/pytorch_repros/bad_aten_mode.py", line 49, in test_py_diagnose_emits_error
    torch.onnx.export(
  File "/github/pytorch-dev2/torch/onnx/__init__.py", line 305, in export
    return utils.export(model, args, f, export_params, verbose, training,
  File "/github/pytorch-dev2/torch/onnx/utils.py", line 118, in export
    _export(model, args, f, export_params, verbose, training, input_names, output_names,
  File "/github/pytorch-dev2/torch/onnx/utils.py", line 719, in _export
    _model_to_graph(model, args, verbose, input_names,
  File "/github/pytorch-dev2/torch/onnx/utils.py", line 503, in _model_to_graph
    graph = _optimize_graph(graph, operator_export_type,
  File "/github/pytorch-dev2/torch/onnx/utils.py", line 232, in _optimize_graph
    graph = torch._C._jit_pass_onnx(graph, operator_export_type)
  File "/github/pytorch-dev2/torch/onnx/__init__.py", line 354, in _run_symbolic_function
    return utils._run_symbolic_function(*args, **kwargs)
  File "/github/pytorch-dev2/torch/onnx/utils.py", line 1057, in _run_symbolic_function
    symbolic_fn = _find_symbolic_in_registry(domain, op_name, opset_version, operator_export_type)
  File "/github/pytorch-dev2/torch/onnx/utils.py", line 1011, in _find_symbolic_in_registry
    return sym_registry.get_registered_op(op_name, domain, opset_version)
  File "/github/pytorch-dev2/torch/onnx/symbolic_registry.py", line 129, in get_registered_op
    raise RuntimeError(msg)
RuntimeError: Exporting the operator diagonal to ONNX opset version 9 is not supported. Support for this operator was added in version 13, try exporting with this version.
```

Isn't that the expected behavior?

Comment (thiagocrepaldi) [reply torch/onnx/utils.py:1850]: The difference from the original behavior (after aten fallback but before the new registry was similar, but `RuntimeError` was used instead

Comment (thiagocrepaldi) [reply torch/onnx/utils.py:1850]: Nope, there was a failure in the logic. Thanks for catching this

**Comment by thiagocrepaldi** [comment]
@BowenBao all tests are passing now

File: test/onnx/test_pytorch_onnx_no_runtime.py [hunk 5]
```diff
@@ -436,6 +439,7 @@ def forward(self, x):
                 )
             ],
             operator_export_type=torch.onnx.OperatorExportTypes.ONNX_ATEN_FALLBACK,
+            verbose=True,
```
Comment (BowenBao) [inline test/onnx/test_pytorch_onnx_no_runtime.py:442]: nit: verbose

**Review by BowenBao — APPROVED** [review]
LGTM 🚢

File: test/onnx/test_pytorch_onnx_no_runtime.py (hunk 5 above)
Comment (thiagocrepaldi) [reply test/onnx/test_pytorch_onnx_no_runtime.py:442]: removed

**Comment by thiagocrepaldi** [comment]
https://github.com/pytorch/pytorch/pull/84382 was the base PR that replaced the "old" implementation for onnx symbolic registry, leading to a regression on 1.13 that was fixed by this PR

**Pull request closed without merging**